craft_providers/bases/buildd.py

```diff
@@ -0,0 +1,520 @@
+                RouteMetric=100
+                UseMTU=true
+                """
+                ).encode()
+            ),
+            file_mode="0644",
+        )
+
+        try:
+            executor.execute_run(
+                ["systemctl", "enable", "systemd-networkd"],
+                capture_output=True,
+                check=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+
+            executor.execute_run(
+                ["systemctl", "restart", "systemd-networkd"],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+        except subprocess.CalledProcessError as error:
+            raise BaseConfigurationError(
+                brief="Failed to setup systemd-networkd.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def _setup_resolved(self, *, executor: Executor, deadline: Optional[float]) -> None:
+        """Configure system-resolved to manage resolve.conf.
+
+        :param executor: Executor for target container.
+        :param timeout_secs: Timeout in seconds.
+        """
+        try:
+            executor.execute_run(
+                [
+                    "ln",
+                    "-sf",
+                    "/run/systemd/resolve/resolv.conf",
+                    "/etc/resolv.conf",
+                ],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+
+            executor.execute_run(
+                ["systemctl", "enable", "systemd-resolved"],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+
+            executor.execute_run(
+                ["systemctl", "restart", "systemd-resolved"],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+        except subprocess.CalledProcessError as error:
+            raise BaseConfigurationError(
+                brief="Failed to setup systemd-resolved.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def _setup_snapd(
+        self, *, executor: Executor, deadline: Optional[float] = None
+    ) -> None:
+        """Install snapd and dependencies and wait until ready.
+
+        :param executor: Executor for target container.
+        :param deadline: Optional time.time() deadline.
+        """
+        try:
+            executor.execute_run(
+                [
+                    "apt-get",
+                    "install",
+                    "-y",
+                    "fuse",
+                    "udev",
```

Comment:
ask around to see if we still need udev :-)

Reply:
I think I hit this myself, but I can run a set of tests without it.

Reply:
Actually, I'll put a TODO here to revisit after LXD provider is back in shape.  And now that I see my TODOs all the time, it won't get lost :rofl: